lib/collection/src/operations/types.rs

```diff
@@ -774,7 +774,7 @@ pub struct VectorParams {
     /// Type of distance function used for measuring distance between vectors
     pub distance: Distance,
     /// Custom params for HNSW index. If none - values from collection configuration are used.
-    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[serde(default, skip_serializing_if = "is_hnsw_diff_empty")]
```

Comment:
Why do we introduce `is_hnsw_diff_empty`?  Is something wrong with `Option::is_none`?

Reply:
The `HnswConfigDiff` may contain no parameters at all. In that case it makes no sense to serialize.

`Option::is_none` does not detect this situation.